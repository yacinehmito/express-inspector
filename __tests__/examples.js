const fs = require("fs");
const path = require("path");

jest.mock("express-inspector", () => {
  const inspector = jest.requireActual("..");

  // We want the trace function to use jest.setMock as the replacer
  // because jest tests don't have a module cache to hack
  const trace = new Proxy(inspector.trace, {
    apply(target, context, args) {
      args = args.slice(0);
      const options = Object.assign({}, args[0], {
        replacer: jest.setMock
      });
      args[0] = options;
      return Reflect.apply(target, context, args);
    }
  });

  // We don't want inspect to write to stdout so we overwrite the logger
  // and set it before each test
  const inspect = new Proxy(inspector.inspect, {
    apply(target, context, args) {
      args = args.slice(0);
      const options = Object.assign({}, args[1], {
        logger: mockLogger
      });
      args[1] = options;
      return Reflect.apply(target, context, args);
    }
  });

  return Object.assign({}, inspector, { trace, inspect });
});

const undefinedLogger = () => {
  throw new Error("Logger should be set");
};

let mockLogger = undefinedLogger;

function runExample(example) {
  let output;
  mockLogger = logged => {
    output = logged;
  };
  require(path.join(__dirname, "../examples", example));
  mockLogger = undefinedLogger;
  return output;
}

jest.mock("express", () => {
  const express = jest.requireActual("express");

  // We don't want to actually start the server
  express.application.listen = () => {};

  return express;
});

fs.readdirSync(path.join(__dirname, "../examples")).map(example => {
  describe(`Example ${example}`, () => {
    const output = runExample(example);
    it("gives the same output as before", () => {
      expect(output).toMatchSnapshot();
    });
  });
});
