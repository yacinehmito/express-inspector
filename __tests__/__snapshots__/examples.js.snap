// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Example simple gives the same output as before 1`] = `
"Route /:name               [2mexamples/simple/index.js:7:5[0m
  GET / <anonymous>        [2mexamples/simple/index.js:7:5[0m
"
`;

exports[`Example standard gives the same output as before 1`] = `
"Route /foos                [2mexamples/standard/app.js:7:5[0m
  POST / <anonymous>       [2mexamples/standard/app.js:7:5[0m
Route /foos/:id            [2mexamples/standard/app.js:19:1[0m
  GET / <anonymous>        [2mexamples/standard/app.js:20:1[0m
  DELETE / <anonymous>     [2mexamples/standard/app.js:23:7[0m
Router /bars               [2mexamples/standard/app.js:27:5[0m
  Route /                  [2mexamples/standard/router.js:5:8[0m
    GET / <anonymous>      [2mexamples/standard/router.js:5:8[0m
  Route /:id               [2mexamples/standard/router.js:9:8[0m
    GET / <anonymous>      [2mexamples/standard/router.js:9:8[0m
"
`;
